kubernetes/audit-trail-deployment.yaml

```diff
@@ -0,0 +1,104 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: audit-trail-gw
+spec:
+  # TODO adjust after merging
+  replicas: 1
+  revisionHistoryLimit: 3
+  selector:
+    matchLabels:
+      app: audit-trail-gw
+  template:
+    metadata:
+      labels:
+        app: audit-trail-gw
+    spec:
+      initContainers:
+        - name: wait-for-hornet
+          image: busybox:1.28
+          args:
+            - /bin/sh
+            - -c
+            - >
+              set -x;
+              until nslookup $IOTA_HORNET_NODE | sed -e 's/[^/]*\/\/\([^@]*@\)\?\([^:/]*\).*/\2/';
+              do 
+              echo waiting for Hornet;
+              sleep 2;
+              done
+          env:
+            - name: "IOTA_HORNET_NODE"
+              valueFrom:
+                configMapKeyRef:
+                  name: integration-service-api-config
+                  key: IOTA_HORNET_NODE
+
+        - name: wait-for-mongo
+          image: busybox:1.28
+          args:
+            - /bin/sh
+            - -c
+            - >
+              set -x;
+              until nslookup "mongodb-service.$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace).svc.cluster.local";
+              do 
+              echo waiting for MongoDB;
+              sleep 2;
+              done
+      containers:
+        - name: audit-trail-gw
+          # TODO adjust after services are deployed on docker
+          image: audit-trail-gw
+          imagePullPolicy: Never
+          command: ["node", "dist/index.js", "server"]
+          resources:
+            requests:
+              memory: "100Mi"
+              cpu: "250m"
+            limits:
+              memory: "250Mi"
+              cpu: "500m"
+          readinessProbe:
+            httpGet:
+              path: /info
+              port: 3002
+          env:
+            - name: "PORT"
```

Comment:
No need for PORT section: container start always on same port. Port mapping must be external